### PR TITLE
feat(embedded): provides filter bar visibility setting on embedded dashboard (#21069)

### PR DESCRIPTION
**Pull request by jileeon**
### SUMMARY
Provides filter bar visibility setting on embedded dashboards

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/pull/18741

### CODE REVIEW

**Comment by michael-s-molina** [comment]
Thanks for the contribution @jileeon. 

@lilykuang Do you think it makes sense to add these configurations under `dashboardUiConfig`? Something like:

```
dashboardUiConfig: {
   hideTitle: true,
   filters: {
      visible: true,
      expanded: true,
   }
}
```

**Review by lilykuang — APPROVED** [review]

**Review by lilykuang — COMMENTED** [review]
@michael-s-molina you suggestion will work better.

**Comment by jileeon** [comment]
Hi @michael-s-molina,

I've changed the commit for your suggestion.

Thanks

**Review by michael-s-molina — APPROVED** [review]

**Comment by jileeon** [comment]
Hi, Sorry for the wrong constant values. I've corrected it now.
Please check the source code changes. 

Thanks.

File: superset-embedded-sdk/src/index.ts [hunk 1]
```diff
@@ -34,6 +37,11 @@ export type UiConfigType = {
   hideTitle?: boolean
   hideTab?: boolean
   hideChartControls?: boolean
+  filters?: {
+    [key: string]: boolean | undefined
+    visible?: boolean
+    expanded?: boolean
```
Comment (jayakrishnankk) [inline superset-embedded-sdk/src/index.ts:43]: @jileeon - considering `DASHBOARD_UI_FILTER_CONFIG_URL_PARAM_KEY` only has 2 keys namely `visible` and `expanded` and line number 115 does `DASHBOARD_UI_FILTER_CONFIG_URL_PARAM_KEY[key]=filterConfig[key]`, i don't understand why there is a need for line number 41 (`[key: string]: boolean | undefined`) in the code.

I found this code while I was trying to add some additional capability add some extra url parameters to the embedded superset. But I don't want to add that code if the functionality already exist.

Comment (jileeon) [reply superset-embedded-sdk/src/index.ts:43]: Hi @jayakrishnankk, Sorry for the late reply. 

these code you wrote are just for complying with Typescript lint.
I think there are no need to keep them If they should be changed.

Thanks.

**Pull request merged**